third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSCodePartition.cpp

```diff
@@ -2125,7 +2221,11 @@ void insertAsyncComm(
         auto producerAcquirePoint =
             getSameLevelOp(headConsumer, tmaHeadProducer); // tmaHeadProducer;
         builder.setAsynTaskIdsFromArray(masterChannel->relation.first);
-        builder.setInsertionPoint(producerAcquirePoint);
+        if (producerAcquireForChannelLoop) {
+          builder.setInsertionPoint(producerAcquireForChannelLoop);
+        } else {
+          builder.setInsertionPoint(producerAcquirePoint);
```

Comment:
In case you see accruacy issues, this should be consistent with the `copyLoopScheduleInfo` below. However, I'm refactoring this right now, so I wouldn't worry about it.